Firebend.AutoCrud.Core/ObjectMapping/ObjectMapper.cs

```diff
@@ -0,0 +1,93 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection.Emit;
+
+namespace Firebend.AutoCrud.Core.ObjectMapping
+{
+    /// <summary>
+    /// This mapper class finds the matching properties and copies them from source object to target object. The copy function has IL codes to do this task.
+    /// </summary>
+    public class ObjectMapper : BaseObjectMapper
+    {
+        /// <summary>
+        /// Easy access to mapper functions
+        /// </summary>
+        public static ObjectMapper Instance { get; protected set; } = new ObjectMapper();
+
+        private ObjectMapper() { }
+
+        /// <summary>
+        /// This dictionary keeps the convert functions for objects by auto generated names
+        /// </summary>
+        private readonly Dictionary<string, DynamicMethod> _del = new Dictionary<string, DynamicMethod>();
+
+        /// <summary>
+        /// This function creates the mappings between objects and store the mappings in the private dictionary
+        /// </summary>
+        /// <param name="source">The type of the source object</param>
+        /// <param name="target">The type of the target object</param>
+        /// <param name="propertiesToIgnore">These string parameters will be ignored during matching process</param>
+        /// <exception cref="InvalidOperationException">The Invalid Operation Exception will be thrown if it can't find the given property in source/target objects.</exception>
+        protected override string MapTypes(Type source, Type target, params string[] propertiesToIgnore)
+        {
+            var key = GetMapKey(source, target);
+
+            if (propertiesToIgnore.Length > 0)
```

Comment:
i think this is a step in the right direction; however, if you keep calling the function with the same parameters to ignore, it will keep generating a new guid, thus reducing the effectiveness of caching and compiling the mapper. could we concatenate the string array to the key?

Reply:
Hi @JacobAtchley, I've sent another commit regarding your comment.